app/src/main/java/com/github/se/project/ui/components/DisplayLessons.kt

```diff
@@ -0,0 +1,110 @@
+import androidx.compose.ui.unit.dp
+import com.github.se.project.model.lesson.Lesson
+import com.github.se.project.model.lesson.LessonStatus
+
+@Composable
+fun DisplayLessons(
+    modifier: Modifier = Modifier,
+    lessons: List<Lesson>,
+    statusFilter: LessonStatus? = null, // Option to filter by status
+    isTutor: Boolean
+) {
+  val filteredLessons =
+      statusFilter?.let { lessons.filter { lesson -> lesson.status == it } } ?: lessons
+
+  LazyColumn(modifier = modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(4.dp)) {
+    itemsIndexed(filteredLessons) { index, lesson ->
+      Card(
+          modifier =
+              Modifier.fillMaxWidth()
+                  .padding(vertical = 2.dp)
+                  .testTag("lessonCard_$index"), // Tag for each lesson card
+          colors =
+              CardDefaults.cardColors(
+                  containerColor =
+                      if (LessonStatus.COMPLETED == lesson.status) {
+                        MaterialTheme.colorScheme.secondaryContainer
+                      } else if (LessonStatus.PENDING == lesson.status ||
+                          LessonStatus.REQUESTED == lesson.status) {
+                        if (isSystemInDarkTheme()) {
+                          MaterialTheme.colorScheme.primaryContainer
+                        } else {
+                          Color(0xFFFEDF89)
+                        }
+                      } else {
+                        MaterialTheme.colorScheme.surfaceBright
+                      })) {
+            Column(
+                modifier =
+                    Modifier.padding(16.dp)
+                        .fillMaxWidth()
+                        .testTag("lessonContent_$index"), // Tag for the lesson content
+                verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                  Text(
+                      text = lesson.title,
+                      style = MaterialTheme.typography.titleMedium,
+                      modifier = Modifier.testTag("lessonTitle_$index") // Tag for the lesson title
+                      )
+
+                  if (LessonStatus.COMPLETED == lesson.status ||
+                      LessonStatus.CONFIRMED == lesson.status) {
+                    Text(
+                        text =
+                            if (isTutor) "With: ${lesson.studentUid}"
+                            else "With: ${lesson.tutorUid}",
+                        style = MaterialTheme.typography.bodyLarge,
+                        modifier =
+                            Modifier.testTag(
+                                "lessonParticipant_$index") // Tag for the tutor/student info
+                        )
+                  }
+
+                  Row(
+                      modifier = Modifier.fillMaxWidth(),
+                      horizontalArrangement = Arrangement.SpaceBetween) {
+                        Text(
+                            text = "Price: ${lesson.price}.-",
```

Comment:
For the price I think it is quite confusing if the price is 0 (which is the base value) when the lesson is pendign and so has no price yet. Maybe we could add an if statement such that if the price is 0.0 we print UNDEF of something like this (assuming that tutor will never give free courses)